lib/dry/transaction/stack.rb

```diff
@@ -0,0 +1,22 @@
+module Dry
+  module Transaction
+    # @api private
+    class Stack
+      LOOPBACK = :itself.to_proc.freeze
+
+      def initialize(steps)
+        @stack = compile(steps)
+      end
+
+      def call(m)
+        @stack.(m)
+      end
+
+      def compile(steps)
```

Comment:
Oh, I reckon it'd be good to drop this under a `private`, if you don't mind :)

Reply:
Sure thing